Tests/test_align_substitution_matrices.py

```diff
@@ -2412,6 +2412,28 @@ def test_loading(self):
             except Exception:
                 self.fail(f"Failed to load subsitution matrix '{name}'")
 
+    def test_reading(self):
+        """Confirm matrix reading works with filename or handle."""
+        matrix_name = "BLOSUM62"
+        test_path = os.path.dirname(__file__)
+        parent_dir = os.path.dirname(test_path)
+        sub_mx_dir = os.path.join(
+            parent_dir, "Bio", "Align", "substitution_matrices", "data"
+        )
+        matrix_path = os.path.join(sub_mx_dir, matrix_name)
+
+        fname_matrix = substitution_matrices.read(matrix_path)
+        self.assertEqual(fname_matrix["A"]["A"], 4.0)
```

Comment:
It may not matter much here, as 4.0 is an integer and may be stored without roundoff, but in general it is better to use `self.assertAlmostEqual` for floating point numbers.